mas-foundation/src/main/java/com/ca/mas/core/oauth/CodeVerifierCache.java

```diff
@@ -8,46 +8,51 @@
 
 package com.ca.mas.core.oauth;
 
+import com.ca.mas.core.storage.sharedstorage.SharedPreferencesUtil;
+
 /**
- * Temporary cache to store code verifier
+ *  Shared preference implementation to store code verifier
  */
 public class CodeVerifierCache {
 
     private static CodeVerifierCache instance = new CodeVerifierCache();
-
-    private String state;
-    private String codeVerifier;
+    private SharedPreferencesUtil prefUtil = null;
+    private String mState = "##default-state##";
```

Comment:
can you make this static final;
private static final String DEFAULT_STATE = "##default-state##";